### PR TITLE
[ADF-1403] Upload Dialog - Delete files event

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell-ng2/app/components/files/files.component.ts [hunk 1]
```diff
@@ -161,7 +162,7 @@ export class FilesComponent implements OnInit {
         this.errorMessage = null;
     }
 
-    onFileUploadComplete(event: FileUploadCompleteEvent) {
+    onFileUploadEvent(event: FileUploadCompleteEvent | FileUploadDeleteEvent) {
```
Comment (DenisVuyka) [inline demo-shell-ng2/app/components/files/files.component.ts]: They both extend FileUploadEvent, can you use it instead the "OR" clause?

File: ng2-components/ng2-alfresco-core/src/services/upload.service.ts [hunk 2]
```diff
@@ -183,7 +184,7 @@ export class UploadService {
             emitter.emit({ value: data });
         })
         .catch(err => {
-            this.onUploadError(file, err);
+            return err;
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/upload.service.ts]: the method is called "getUploadPromise", does "return err" result in a promise?

File: ng2-components/ng2-alfresco-upload/index.ts [hunk 3]
```diff
@@ -47,7 +45,7 @@ export const UPLOAD_DIRECTIVES: any[] = [
 ];
 
 export const UPLOAD_PROVIDERS: any[] = [
-    FileUploadService
+
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-upload/index.ts]: What is the reason behind this empty array then?

File: ng2-components/ng2-alfresco-upload/src/components/file-uploading-dialog.component.ts [hunk 4]
```diff
@@ -31,35 +32,51 @@ export class FileUploadingDialogComponent implements OnInit, OnDestroy {
...
     isDialogMinimized: boolean = false;
-    uploadFilesCompleted: boolean = false;
 
     private listSubscription: Subscription;
     private counterSubscription: Subscription;
     private fileUploadSubscription: Subscription;
+    private errorSubscription: Subscription;
 
     constructor(
+        translateService: AlfrescoTranslationService,
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-upload/src/components/file-uploading-dialog.component.ts]: This approach has been deprecated. All i18n resources are initialised on per-module basis now. Please remove "translateService.addTranslationFolder" from this file.

File: ng2-components/ng2-alfresco-core/src/services/upload.service.ts (hunk 2 above)
Comment (pionnegru) [inline ng2-components/ng2-alfresco-core/src/services/upload.service.ts]: `.catch()` returns a promise. The only problem I see with this statement is that a later `.then()` will be trigger and not a `.catch()`. We should `throw err`.

**Pull request merged**